pkg/flow/states/foreach.go

```diff
@@ -270,33 +272,33 @@ func (logic *forEachLogic) processActionResults(ctx context.Context, children []
 	if results.ActionID != id {
 		return nil, derrors.NewInternalError(errors.New("incorrect child action ID"))
 	}
-
-	logic.Log(ctx, "Child '%s' returned.", id)
+	logic.AddAttribute("loop-index", fmt.Sprintf("%d", idx))
```

Comment:
What does this achieve? It seems strange to me to attach it to the `logic` here because the `logic` is running multiple iterators in parallel.